hazelcast/include/hazelcast/util/Future.h

```diff
@@ -94,7 +94,11 @@ namespace hazelcast {
                 }
                 time_t endTime = time(NULL) + timeInSeconds;
                 while (!(resultReady || exceptionReady) && endTime > time(NULL)) {
-                    conditionVariable.waitFor(mutex, endTime - time(NULL));
+                    if (!conditionVariable.waitFor(mutex, endTime - time(NULL))) {
+                        std::ostringstream out;
+                        out << "Future::get(timeInSeconds) timed out. Timeout: " << timeInSeconds;
+                        hazelcast::util::ILogger::getLogger().warning(out.str());
```

Comment:
I think we should not log here. We will tell the timeout to user via TimeoutException.